XRTK-Core/Packages/com.xrtk.core/Inspectors/Profiles/InputSystem/Controllers/BaseMixedRealityControllerDataProviderProfileInspector.cs

```diff
@@ -168,17 +171,86 @@ public override void OnInspectorGUI()
         {
             RenderHeader("This profile defines all of the controllers and their mappings to use. Additional platform settings can also be available as well.");
 
+            showMappingProfiles = EditorGUILayoutExtensions.FoldoutWithBoldLabel(showMappingProfiles, new GUIContent("Controller Mapping Profiles"), true);
+            if (showMappingProfiles)
+            {
+                EditorGUILayout.Space();
+                selectedMappingsViewModeTab = GUILayout.Toolbar(selectedMappingsViewModeTab, new string[] { "Simple", "Advanced" });
+                switch (selectedMappingsViewModeTab)
+                {
+                    case 0:
+                        DrawSimpleControllerMappingProfilesView();
+                        break;
+                    case 1:
+                        DrawAdvancedControllerMappingProfilesView();
+                        break;
+                }
+            }
+
+            EditorGUILayout.Space();
+        }
+
+        private void DrawAdvancedControllerMappingProfilesView()
+        {
             serializedObject.Update();
+            mappingProfileList.DoLayoutList();
+            serializedObject.ApplyModifiedProperties();
+        }
 
-            showMappingProfiles = EditorGUILayout.Foldout(showMappingProfiles, new GUIContent("Controller Mapping Profiles"), true);
+        private void DrawSimpleControllerMappingProfilesView()
+        {
+            if (controllerButtonStyle == null)
+            {
+                controllerButtonStyle = new GUIStyle("LargeButton")
+                {
+                    imagePosition = ImagePosition.ImageAbove,
+                    fontStyle = FontStyle.Bold,
+                    stretchHeight = true,
+                    stretchWidth = true,
+                    wordWrap = true,
+                    fontSize = 10,
+                };
+            }
 
-            if (showMappingProfiles)
+            bool appliedModifications = false;
+            serializedObject.Update();
+
+            for (int i = 0; i < controllerMappingProfiles?.arraySize; i++)
             {
-                mappingProfileList.DoLayoutList();
+                var targetObjectReference = controllerMappingProfiles.GetArrayElementAtIndex(i)?.objectReferenceValue;
```

Comment:
why would an element ever be null?

Reply:
if it is, then shouldn't we just remove it from the array?

Reply:
It can be null since the UI in advanced mode has that "+" Button to add new entries to the list. If those entries stay empty we get null. But yeah, probably better to just cleanup them in that case.